pyaerocom/io/read_eea_aqerep_base.py

```diff
@@ -0,0 +1,643 @@
+import pandas as pd
+from tqdm import tqdm
+
+from pyaerocom import const
+from pyaerocom.exceptions import TemporalResolutionError
+from pyaerocom.stationdata import StationData
+from pyaerocom.tstype import TsType
+from pyaerocom.ungriddeddata import UngriddedData
+
+from pyaerocom.io.helpers import get_country_name_from_iso
+from pyaerocom.io.helpers_units import unitconv_concx_to_vmrx
+from pyaerocom.io.readungriddedbase import ReadUngriddedBase
+
+
+class ReadEEAAQEREPBase(ReadUngriddedBase):
+    """Class for reading EEA AQErep data
+
+    Extended class derived from  low-level base class
+    :class:`ReadUngriddedBase` that contains some more functionality.
+
+    Note
+    ----
+    Currently only single variable reading into an :class:`UngriddedData`
+    object is supported.
+    """
+    #: Mask for identifying datafiles
+    _FILEMASK = '*.csv'
+
+    #: Version log of this class (for caching)
+    __version__ = '0.05'
+
+    #: Column delimiter
+    FILE_COL_DELIM = ','
+
+    #: Name of the dataset (OBS_ID)
+    DATA_ID = ''  # change this since we added more vars?
+
+    #: List of all datasets supported by this interface
+    SUPPORTED_DATASETS = [DATA_ID]
+
+    #: There is no global ts_type but it is specified in the data files...
+    TS_TYPE = 'variable'
+
+    #: sampling frequencies found in data files
+    TS_TYPES_FILE = {
+        'hour': 'hourly',
+        'day': 'daily',
+    }
+
+    #: Dictionary specifying values corresponding to invalid measurements
+    #: there's no value for NaNs in this data set. It uses an empty string
+    NAN_VAL = {}
+
+    #: Dictionary specifying the file column names (values) for each Aerocom
+    #: variable (keys)
+    # There's only one variable in each file named concentration
+    VAR_NAMES_FILE = {}
+    VAR_NAMES_FILE['concso2'] = 'concentration'
+    VAR_NAMES_FILE['conco3'] = 'concentration'
+    VAR_NAMES_FILE['concno2'] = 'concentration'
+    VAR_NAMES_FILE['concco'] = 'concentration'
+    VAR_NAMES_FILE['concno'] = 'concentration'
+    VAR_NAMES_FILE['vmro3'] = 'concentration'
+    VAR_NAMES_FILE['concpm10'] = 'concentration'
+    VAR_NAMES_FILE['concpm25'] = 'concentration'
+
+    #: units of variables in files (needs to be defined for each variable supported)
+    VAR_UNITS_FILE = {
+        'µg/m3': 'ug m-3',
+        'mg/m3': 'mg m-3',
+        'ppb': 'ppb'
+    }
+
+    #: file masks for the data files
+    FILE_MASKS = {}
+    FILE_MASKS['concso2'] = '**/*_1_*_timeseries.csv'
+    FILE_MASKS['concpm10'] = '**/*_5_*_timeseries.csv'
+    FILE_MASKS['conco3'] = '**/*_7_*_timeseries.csv'
+    FILE_MASKS['vmro3'] = '**/*_7_*_timeseries.csv'
+    FILE_MASKS['concno2'] = '**/*_8_*_timeseries.csv'
+    FILE_MASKS['concco'] = '**/*_10_*_timeseries.csv'
+    FILE_MASKS['concno'] = '**/*_38_*_timeseries.csv'
+    FILE_MASKS['concpm25'] = '**/*_6001_*_timeseries.csv'
+
+    # conversion factor between concX and vmrX
+    CONV_FACTOR = {}
+    CONV_FACTOR['vmro3'] = np.float_(0.5)
```

Comment:
You have conv factor here and in the function in helpers_units. Can we delete the function in helpers_units then?

Reply:
I did not want this in the helper_units.py, but when I deleted that, I could not use AUX_REQUIRES anymore since one can only live with the other it seems. But I thought it would be good to follow the practice to use AUX_REQUIRES.
I was basically waiting on your comment on this :-)

Reply:
Okay, but if we need that function, then it should at least raise an Exception. I made a code suggestion, which you could just commit directly.

Reply:
agreed

Reply:
I have committed your suggestion